corehq/apps/domain/views.py

```diff
@@ -2506,23 +2506,31 @@ def page_context(self):
             'repeater_type': self.repeater_type,
         }
 
+    def initialize_repeater(self):
+        return self.repeater_class()
+
     def make_repeater(self):
-        repeater = self.repeater_class(
-            domain=self.domain,
-            url=self.add_repeater_form.cleaned_data['url'],
-            use_basic_auth=self.add_repeater_form.cleaned_data['use_basic_auth'],
-            username=self.add_repeater_form.cleaned_data['username'],
-            password=self.add_repeater_form.cleaned_data['password'],
-            format=self.add_repeater_form.cleaned_data['format']
-        )
```

Comment:
I broke this method to make this more flexible so that this class can be extended easily. For ex extracted out initialize_repeater so that it can be initialised with an existing repeater when editing and not always a new repeater. Similarly extracted post_save